classes/controllers/grid/users/reviewer/PKPReviewerGridHandler.inc.php

```diff
@@ -29,32 +29,29 @@
 use PKP\core\PKPRequest;
 use PKP\core\PKPServices;
 use PKP\db\DAORegistry;
+use PKP\facades\Locale;
 use PKP\linkAction\LinkAction;
 use PKP\linkAction\request\AjaxModal;
 use PKP\log\SubmissionLog;
 use PKP\emailtemplate\EmailTemplate;
 use PKP\mail\Mailable;
 use PKP\mail\mailables\MailReviewerReinstated;
 use PKP\mail\mailables\MailReviewerUnassigned;
-use PKP\mail\SubmissionMailTemplate;
+use PKP\mail\traits\Sender;
 use PKP\notification\PKPNotification;
 use PKP\notification\PKPNotificationManager;
 use PKP\security\authorization\internal\ReviewAssignmentRequiredPolicy;
 use PKP\security\authorization\internal\ReviewRoundRequiredPolicy;
 use PKP\security\authorization\WorkflowStageAccessPolicy;
 use PKP\security\Role;
+use PKP\user\User;
+use Symfony\Component\Mailer\Exception\TransportException;
 
 // FIXME: Add namespacing
 import('lib.pkp.controllers.grid.users.reviewer.ReviewerGridCellProvider');
-
-use PKP\user\User;
-use ReinstateReviewerForm;
-use ReviewerGridCellProvider;
-
 import('lib.pkp.controllers.grid.users.reviewer.ReviewerGridRow');
+use ReviewerGridCellProvider;
 use ReviewerGridRow;
-use Swift_TransportException;
-use UnassignReviewerForm;
 
 class PKPReviewerGridHandler extends GridHandler
```

Comment:
I got the following fatal error when I tried to read and confirm a submitted review. Steps to reproduce:

1. Login as reviewer
2. Confirm and complete review
3. Login as editor, find review assignment
4. Click Read Review > Confirm

```
Next Illuminate\Database\QueryException: SQLSTATE[22007]: Invalid datetime format: 7 ERROR:  invalid input syntax for type timestamp: "2022-03-28 16:Mar:th" (SQL: insert into "item_views" ("date_last_viewed", "assoc_type", "assoc_id", "user_id") values (2022-03-28 16:Mar:th, 516, 26, 3) on conflict ("assoc_type", "assoc_id", "user_id") do update set "date_last_viewed" = "excluded"."date_last_viewed", "assoc_type" = "excluded"."assoc_type", "assoc_id" = "excluded"."assoc_id", "user_id" = "excluded"."user_id") in lib/pkp/lib/vendor/laravel/framework/src/Illuminate/Database/Connection.php:671
Stack trace:
#0 lib/pkp/lib/vendor/laravel/framework/src/Illuminate/Database/Connection.php(631): Illuminate\Database\Connection->runQueryCallback('insert into "it...', Array, Object(Closure))
#1 lib/pkp/lib/vendor/laravel/framework/src/Illuminate/Database/Connection.php(496): Illuminate\Database\Connection->run('insert into "it...', Array, Object(Closure))
#2 lib/pkp/lib/vendor/staudenmeir/laravel-upsert/src/Query/Builder.php(92): Illuminate\Database\Connection->affectingStatement('insert into "it...', Array)
#3 lib/pkp/classes/db/DAO.inc.php(186): Staudenmeir\LaravelUpsert\Query\Builder->upsert(Array, Array)
#4 lib/pkp/classes/views/ViewsDAO.inc.php(39): PKP\db\DAO->replace('item_views', Array, Array)
#5 lib/pkp/classes/controllers/grid/users/reviewer/PKPReviewerGridHandler.inc.php(647): PKP\views\ViewsDAO->recordView(516, 26, 3)
#6 controllers/grid/users/reviewer/ReviewerGridHandler.inc.php(42): PKP\controllers\grid\users\reviewer\PKPReviewerGridHandler->reviewRead(Array, Object(APP\core\Request))
#7 [internal function]: ReviewerGridHandler->reviewRead(Array, Object(APP\core\Request))
#8 lib/pkp/classes/core/PKPRouter.inc.php(465): call_user_func(Array, Array, Object(APP\core\Request))
#9 lib/pkp/classes/core/PKPComponentRouter.inc.php(288): PKP\core\PKPRouter->_authorizeInitializeAndCallRequest(Array, Object(APP\core\Request), Array)
#10 lib/pkp/classes/core/Dispatcher.inc.php(163): PKP\core\PKPComponentRouter->route(Object(APP\core\Request))
#11 lib/pkp/classes/core/PKPApplication.inc.php(375): PKP\core\Dispatcher->dispatch(Object(APP\core\Request))
#12 index.php(70): PKP\core\PKPApplication->execute()
#13 {main}
  thrown in lib/pkp/lib/vendor/laravel/framework/src/Illuminate/Database/Connection.php on line 671
```

Reply:
That was related to the recent changes of date & time formatting

Reply:
This should've fixed it: https://github.com/pkp/pkp-lib/commit/acbcc6c51081510f0b6728afca6ba9a7cdb3b0a6